config/config.go

```diff
@@ -105,6 +107,26 @@ type BaggageRestrictionsConfig struct {
 	RefreshInterval time.Duration `yaml:"refreshInterval"`
 }
 
+// ThrottlerConfig configures the throttler which can be used to throttle the
+// rate at which the client may send debug requests.
+type ThrottlerConfig struct {
+	// HostPort of jaeger-agent's credit server.
+	HostPort string `yaml:"hostPort"`
+	// RefreshInterval controls how often the throttler will poll jaeger-agent
+	// for more throttling credits.
+	RefreshInterval time.Duration `yaml:"refreshInterval"`
+	// SynchronousInitialization determines whether or not the throttler should
+	// synchronously fetch credits from the agent when an operation is seen for
+	// the first time. This should be set to true if the client will be used by
+	// a short lived service that needs to ensure that credits are fetched
+	// upfront such that sampling or throttling occurs.
+	SynchronousInitialization bool `yaml:"synchronousInitialization"`
+	// MaximumBalance limits the maximum number of credits the throttler will
+	// accumulate on a per-operation basis before it will stop requesting more
+	// credits. To leave it unbounded, leave this as zero.
+	MaximumBalance float64 `yaml:"maximumBalance"`
```

Comment:
I am not a fan of this variable. The agent should handle all allocations and determining the maximum. Also, looking at the code, it looks like you have if maximumBalance is allocated, then the client doesn't make a request to the agent which means the operation will fall out of the agent's memory due to the TTL.

Reply:
True. I will remove.